src/sqlfluff/core/parser/grammar/sequence.py

```diff
@@ -484,109 +482,107 @@ def match(
-        if (
-            not content_match.matched_slice.stop == _end_idx
-            and self.parse_mode == ParseMode.STRICT
-        ):
-            return MatchResult.empty_at(idx)
+            content_match = super().match(segments, content_start_idx, ctx)
 
-        # What's between the final match and the content. Hopefully just gap?
-        intermediate_slice = slice(
-            # NOTE: Assumes that brackets are always of size 1.
-            content_match.matched_slice.stop,
-            bracketed_match.matched_slice.stop - 1,
-        )
-        if not self.allow_gaps and not is_zero_slice(intermediate_slice):
-            # NOTE: In this clause, content_match will never have matched. Either
-            # we're in STRICT mode, and would have exited in the `return` above,
-            # or we're in GREEDY mode and the `super().match()` will have already
-            # claimed the whole sequence with nothing left. This clause is
-            # effectively only accessible in a bracketed section which doesn't
-            # allow whitespace but nonetheless has some, which is fairly rare.
-            expected = str(self._elements)
-            # Whatever is in the gap should be marked as an UnparsableSegment.
-            child_match = MatchResult(
-                intermediate_slice,
-                UnparsableSegment,
-                segment_kwargs={"expected": expected},
+        # Get position after content for end bracket check
+        gap_start = end_bracket_idx = content_match.matched_slice.stop
+
+        # Skip whitespace if allowed
+        if self.allow_gaps:
+            end_bracket_idx = skip_start_index_forward_to_code(
+                segments, end_bracket_idx
             )
-            content_match = content_match.append(child_match)
 
-        # We now have content and bracketed matches. Depending on whether the intent
-        # is to wrap or not we should construct the response.
-        _content_matches: tuple[MatchResult, ...]
-        if content_match.matched_class:
-            _content_matches = bracketed_match.child_matches + (content_match,)
-        else:
-            _content_matches = (
-                bracketed_match.child_matches + content_match.child_matches
+        # If there's a gap, add it as a child match
+        child_matches: tuple[MatchResult, ...] = (start_match,)
+        if not is_zero_slice(content_match.matched_slice):
+            if content_match.matched_class:
+                child_matches += (content_match,)
+            else:
+                child_matches += content_match.child_matches
+        if end_bracket_idx > gap_start:
+            gap_match = MatchResult(
+                matched_slice=slice(gap_start, end_bracket_idx),
+                matched_class=None,
+                segment_kwargs={},
+                insert_segments=(),
+                child_matches=(),
+            )
+            child_matches += (gap_match,)
```

Comment:
We add a gap match at the end, and not at the start. That feels inconsistent, but I'm also not sure about the implications of that given the test suite still passes. Can you review what happens if we remove this, or what happens if we also put one at the start?

I'm assuming there _is_ a reason, and if you work out what that is, I think we should add a comment to explain that.